### PR TITLE
langgraph: add support for store in ToolNode

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

File: libs/langgraph/langgraph/utils/runnable.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ class StrEnum(str, enum.Enum):
         sys.intern("store"),
         (BaseStore, "BaseStore", inspect.Parameter.empty),
         CONFIG_KEY_STORE,
-        inspect.Parameter.empty,
+        None,
```
Comment (vbarda) [inline libs/langgraph/langgraph/utils/runnable.py:74]: @nfcampos i am a bit concerned about this one, but without this change you would get
```python
ValueError: Missing required config key '__pregel_store' for 'tools'.
```
when running `tool_node.invoke()` without passing store kwarg

File: libs/langgraph/langgraph/prebuilt/tool_node.py [hunk 2]
```diff
@@ -391,23 +425,78 @@ def __init__(self, field: Optional[str] = None) -> None:
...
+        from langchain_core.tools import tool
+
+        from langgraph.store.memory import InMemoryStore
+        from langgraph.prebuilt import InjectedStore, ToolNode
+
+        store = InMemoryStore()
+        store.put(("values",), "foo", {"bar": 2})
+
+        @tool
+        def store_tool(x: int, my_store: Annotated[Any, InjectedStore()]) -> str:
```
Comment (hinthornw) [inline libs/langgraph/langgraph/prebuilt/tool_node.py:453]: Does it have to be `Any`?
Thoughts on it being

```
Annotated[BaseStore,  InjectedToolArg]
```

(so we don't require duplicate annotation type things?)

File: libs/langgraph/langgraph/utils/runnable.py (hunk 1 above)
Comment (hinthornw) [reply libs/langgraph/langgraph/utils/runnable.py:74]: Right now I think raising an error is good actually?  

Would be better to have a specific user-facing one though. Maybe even handle better in the tool node? (like saying "please invoke within a graph where you've provided a store" or something of that nature

File: libs/langgraph/langgraph/prebuilt/tool_node.py [hunk 3]
```diff
@@ -223,8 +232,6 @@ def _inject_state(
             BaseModel,
         ],
     ) -> ToolCall:
-        if tool_call["name"] not in self.tools_by_name:
-            return tool_call
         state_args = _get_state_args(self.tools_by_name[tool_call["name"]])
```
Comment (hinthornw) [inline libs/langgraph/langgraph/prebuilt/tool_node.py:235]: We should be doing this once in the constructor and caching so we don't have to do all this meta-programming every time the node is called

Comment (vbarda) [reply libs/langgraph/langgraph/prebuilt/tool_node.py:235]: nice, good idea

File: libs/langgraph/langgraph/prebuilt/tool_node.py (hunk 2 above)
Comment (vbarda) [reply libs/langgraph/langgraph/prebuilt/tool_node.py:453]: let me improve

Comment (vbarda) [reply libs/langgraph/langgraph/prebuilt/tool_node.py:453]: there is an issue in `core`, will update once fixed up

Comment (hinthornw) [reply libs/langgraph/langgraph/prebuilt/tool_node.py:479]: 🉑

**Review by hinthornw — APPROVED** [review]

**Pull request merged**